civis/cli/_cli_commands.py

```diff
@@ -156,6 +165,55 @@ def _str_table_result(cols, rows):
     return '\n'.join(tb_strs)
 
 
+@click.command(
+    'follow-log',
+    help='Output live log from the most recent job run.' + _FOLLOW_LOG_NOTE)
+@click.argument('id', type=int)
+def jobs_follow_log(id):
+    client = civis.APIClient()
+    runs = client.jobs.list_runs(id, limit=1, order='id', order_dir='desc')
+    if not runs:
+        raise click.ClickException('No runs found for that job ID.')
+    run_id = runs[0].id
+    print('Run ID: ' + str(run_id))
+    _jobs_follow_run_log(id, run_id)
+
+
+@click.command(
+    'follow-run-log',
+    help='Output live run log.' + _FOLLOW_LOG_NOTE)
+@click.argument('id', type=int)
+@click.argument('run_id', type=int)
+def jobs_follow_run_log(id, run_id):
+    _jobs_follow_run_log(id, run_id)
+
+
+def _jobs_follow_run_log(id, run_id):
+    client = civis.APIClient(return_type='raw')
+    seen_max_log_id = 0
+    continue_polling = True
+
+    while continue_polling:
+        response = client.jobs.list_runs_logs(id, run_id,
+                                              last_id=seen_max_log_id)
+        if 'civis-max-id' in response.headers:
+            remote_max_log_id = int(response.headers['civis-max-id'])
+        log_finished = response.headers['civis-cache-control'] != 'no-store'
+        logs = response.json()
+        if logs:
+            seen_max_log_id = max(log['id'] for log in logs)
+            logs.sort(key=operator.itemgetter('createdAt', 'id'))
+        for log in logs:
+            print(' '.join((log['createdAt'], log['message'].rstrip())))
+        # if output is a pipe, write the buffered output immediately:
+        sys.stdout.flush()
+        if seen_max_log_id == remote_max_log_id:
```

Comment:
What does it mean if this condition is met vs not? If it's not met, won't you be stuck in an infinite loop?

Reply:
If it's not met, we'll continue polling.